cachi2/core/models.py

```diff
@@ -0,0 +1,11 @@
+from gc import freeze
+from pydantic.dataclasses import dataclass
+
+
+@dataclass
+class Request:
+    url: str
+    ref: str
+    flags: tuple = ()
```

Comment:
I like your version much better. Thing is, as of now, a Request is not Gomod, since it can have multiple packages. A Package, in its turn, could be a GomodPackage, PipPackage, etc.

I will bring up a discussion about the scope of a Request, if we should keep it as it is or rework it.

Reply:
Ah, yeah that's true. I guess the CLI could create multiple requests, one for each package type:
```python
class PipRequest(Request):
    packages: list[PipPkgSpec]


class GomodRequest(Request)
    packages: list[GomodPkgSpec]
    cgo_disable: bool = False
    # ... other attrs
```
Or just have one Request for all package types combined
```python
class Request:
    packages: list[PkgSpec]
    cgo_disable: bool = false
    gomod_vendor: ...
```

Reply:
I like the approach with multiple requests a little bit more. Let's discuss in following PR.

Reply:
:+1:  to refactor later